stubs/psycopg2/psycopg2/sql.pyi

```diff
@@ -1,50 +1,50 @@
-from _typeshed import Incomplete
-from collections.abc import Iterator
-from typing import Any
+from _typeshed import SupportsIter
+from collections.abc import Iterable, Iterator
+from typing import Any, Generic, TypeVar
+
+from psycopg2._psycopg import connection, cursor
+
+_T = TypeVar("_T")
 
 class Composable:
-    def __init__(self, wrapped) -> None: ...
-    def as_string(self, context) -> str: ...
-    def __add__(self, other) -> Composed: ...
-    def __mul__(self, n) -> Composed: ...
-    def __eq__(self, other) -> bool: ...
-    def __ne__(self, other) -> bool: ...
+    def __init__(self, wrapped: Any) -> None: ...
+    def as_string(self, context: connection | cursor) -> str: ...
+    def __add__(self, other: Composable) -> Composed: ...
+    def __mul__(self, n: int) -> Composed: ...
+    def __eq__(self, other: object) -> bool: ...
+    def __ne__(self, other: object) -> bool: ...
 
 class Composed(Composable):
-    def __init__(self, seq) -> None: ...
+    def __init__(self, seq: Iterable[Composable]) -> None: ...
     @property
     def seq(self) -> list[Composable]: ...
-    def as_string(self, context) -> str: ...
     def __iter__(self) -> Iterator[Composable]: ...
-    def __add__(self, other) -> Composed: ...
-    def join(self, joiner) -> Composed: ...
+    def __add__(self, other: Composable) -> Composed: ...
+    def join(self, joiner: str | SQL) -> Composed: ...
 
 class SQL(Composable):
-    def __init__(self, string) -> None: ...
+    def __init__(self, string: str) -> None: ...
     @property
     def string(self) -> str: ...
-    def as_string(self, context) -> str: ...
-    def format(self, *args, **kwargs) -> Composed: ...
-    def join(self, seq) -> Composed: ...
+    def format(self, *args: Composable, **kwargs: Composable) -> Composed: ...
+    def join(self, seq: SupportsIter[Composable]) -> Composed: ...
```

Comment:
This needs to be `Iterable[Composable]`, not `SupportsIter[Composable]`. The difference is that

* `seq: Iterable[Composable]` means `iter(seq): Iterator[Composable]` (hence `next(iter(seq)): Composable`), while
* `seq: SupportsIter[Composable]` means `iter(seq): Composable`, which is not an iterator.